mrp_bom_attribute_match_semifinished_products/wizard/finished_product_structure_wizard.py

```diff
@@ -0,0 +1,136 @@
+from odoo import _, api, fields, models
+
+
+class FinishedProductStructureWizard(models.TransientModel):
+    _name = "finished.product.structure.wizard"
+
+    finished_product_id = fields.Many2one(
+        comodel_name="product.template",
+        domain="[('finished_product', '=', True)]",
+        required=True,
+    )
+    line_ids = fields.One2many(
+        comodel_name="finished.product.structure.line", inverse_name="structure_id"
+    )
+    attribute_ids = fields.Many2many(
+        comodel_name="product.attribute", compute="_compute_attribute_ids"
+    )
+    need_confirmation = fields.Boolean()
+
+    @api.onchange("finished_product_id")
+    def _onchange_finished_product_id(self):
+        """
+        Update attribute and line attributes
+        by onchange finished product
+        """
+        if self:
+            self._compute_attribute_ids()
```

Comment:
**question**: can this line be removed if you add the `api.depends` below?

Otherwise this should maybe be an `onchange("attribute_ids")`